prime-router/src/main/kotlin/azure/SubmissionsFacade.kt

```diff
@@ -137,16 +140,43 @@ class SubmissionsFacade(
     }
 
     /**
-     * Find a [reportId] for a given [organizationName].
-     * @return the detailed submission history or null if the report was not found
+     * get a single Action associated with this reportId
      */
-    fun findReport(
-        organizationName: String,
-        reportId: UUID,
-    ): DetailedSubmissionHistory? {
-        val submissionId = dbAccess.fetchActionIdForReport(reportId)
-        return if (submissionId != null) findSubmission(organizationName, submissionId)
-        else null
+    fun fetchActionForReportId(reportId: UUID): Action? {
+        return dbAccess.fetchActionForReportId(reportId)
+    }
+
+    /**
+     * get a single Action, given its key
+     */
+    fun fetchAction(actionId: Long): Action? {
+        return dbAccess.fetchAction(actionId)
+    }
+
+    /**
+     * @return true if [claims] authorizes access to this 'receive' [action].  Return
```

Comment:
kdoc missing description.  You only have @ return

Reply:
done